core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -713,19 +713,14 @@ protected QB copyQuery(QB query) throws IOException {
      * @return a new {@link QueryShardContext} based on the base test index and queryParserService
      */
     protected static QueryShardContext createShardContext() {
-        QueryShardContext queryCreationContext = new QueryShardContext(queryShardContext);
-        queryCreationContext.reset();
-        queryCreationContext.parseFieldMatcher(ParseFieldMatcher.STRICT);
-        return queryCreationContext;
+        return new QueryShardContext(queryShardContext);
```

Comment:
are we losing the STRICT bit here? it's important that we use STRICT here, so we make sure that we never output deprecated stuff ourselves. and we test deprecations separately.